cpp/include/raft/spatial/knn/detail/ann_kmeans_balanced.cuh

```diff
@@ -522,51 +643,54 @@ void balancing_em_iters(const handle_t& handle,
 }
 
 /** Randomly initialize cluster centers and then call `balancing_em_iters`. */
-template <typename T>
+template <typename T, typename IdxT, typename LabelT>
 void build_clusters(const handle_t& handle,
```

Comment:
Since this is called by external algorithms (ivf-pq), could you add an assert here as well, to check that `n_rows * dim` can be represented with `IdxT`?